salt/modules/vsphere.py

```diff
@@ -3601,6 +3601,991 @@ def vsan_enable(host, username, password, protocol=None, port=None, host_names=N
+    if config_dict.get('description'):
+        config_spec.description = config_dict.get('description')
+    if config_dict.get('max_mtu'):
+        config_spec.maxMtu = config_dict.get('max_mtu')
+    if config_dict.get('lacp_api_version'):
+        config_spec.lacpApiVersion = config_dict.get('lacp_api_version')
+    if config_dict.get('network_resource_control_version'):
+        config_spec.networkResourceControlVersion = \
+                config_dict.get('network_resource_control_version')
+    if config_dict.get('uplink_names'):
+        if not config_spec.uplinkPortPolicy or \
+           not isinstance(config_spec.uplinkPortPolicy,
+                          vim.DVSNameArrayUplinkPortPolicy):
+
+            config_spec.uplinkPortPolicy = \
+                    vim.DVSNameArrayUplinkPortPolicy()
+        config_spec.uplinkPortPolicy.uplinkPortName = \
+                config_dict['uplink_names']
+
+
+def _apply_dvs_link_discovery_protocol(disc_prot_config, disc_prot_dict):
+    '''
+    Applies the values of the disc_prot_dict dictionary to a link discovery
+    protocol config object (vim.LinkDiscoveryProtocolConfig)
+    '''
+    disc_prot_config.operation = disc_prot_dict['operation']
+    disc_prot_config.protocol = disc_prot_dict['protocol']
+
+
+def _apply_dvs_product_info(product_info_spec, product_info_dict):
+    '''
+    Applies the values of the product_info_dict dictionary to a product info
+    spec (vim.DistributedVirtualSwitchProductSpec)
+    '''
+    if product_info_dict.get('name'):
+        product_info_spec.name = product_info_dict['name']
+    if product_info_dict.get('vendor'):
+        product_info_spec.vendor = product_info_dict['vendor']
+    if product_info_dict.get('version'):
+        product_info_spec.version = product_info_dict['version']
+
+
+def _apply_dvs_capability(capability_spec, capability_dict):
+    '''
+    Applies the values of the capability_dict dictionary to a DVS capability
+    object (vim.vim.DVSCapability)
+    '''
+    if 'operation_supported' in capability_dict:
+        capability_spec.dvsOperationSupported = \
+                capability_dict['operation_supported']
+    if 'port_operation_supported' in capability_dict:
+        capability_spec.dvPortOperationSupported = \
+                capability_dict['port_operation_supported']
+    if 'portgroup_operation_supported' in capability_dict:
+        capability_spec.dvPortGroupOperationSupported = \
+                capability_dict['portgroup_operation_supported']
+
+
+def _apply_dvs_infrastructure_traffic_resources(infra_traffic_resources,
+                                                resource_dicts):
+    '''
+    Applies the values of the resource dictionaries to infra traffic resources,
+    creating the infra traffic resource if required
+    (vim.DistributedVirtualSwitchProductSpec)
+    '''
+    for res_dict in resource_dicts:
+        ress = [r for r in infra_traffic_resources if r.key == res_dict['key']]
+        if ress:
```

Comment:
IMHO `ress` and `res` as variable names is very hard to read.

Reply:
Will fix